locals.tf

```diff
@@ -6,26 +6,134 @@ locals {
   full_build_id = "${var.tfi_codebuild_id == "" ? format("notcb:%s", uuid()) : var.tfi_codebuild_id}"                                                                                                                            #128-bit rfc 4122 v4 UUID
   build_id      = "${substr(element(split(":",local.full_build_id),1), 0, 8)}${substr(element(split(":",local.full_build_id),1), 9, 4)}"                                                                                         #extract node portion of uuid (last 6 octets) for brevity
   resource_name = "${local.name_prefix}-${local.build_id}"
+}
+
+# magic values
+locals {
+  pypi_url          = "https://pypi.org/simple"
+  pip_bootstrap_url = "https://bootstrap.pypa.io/2.6/get-pip.py"
+
+  win_bootstrap_url        = "https://raw.githubusercontent.com/plus3it/watchmaker/develop/docs/files/bootstrap/watchmaker-bootstrap.ps1"
+  win_python_url           = "https://www.python.org/ftp/python/3.6.6/python-3.6.6-amd64.exe"
+  win_git_url              = "https://github.com/git-for-windows/git/releases/download/v2.18.0.windows.1/Git-2.18.0-64-bit.exe"
+  win_download_dir         = "C:\\Users\\Administrator\\Downloads"
+  win_temp_dir             = "C:\\Temp"
+  win_userdata_status_file = "${local.win_temp_dir}\\userdata_status"
+  win_7zip_url             = "https://www.7-zip.org/a/7z1805-x64.exe"
 
-  default_subnet = "${aws_default_subnet.tfi.id}"
+  lx_temp_dir             = "/tmp"
+  lx_userdata_status_file = "${local.lx_temp_dir}/userdata_status"
+  lx_builder_user         = "ubuntu"
 }
 
-# place to put the ami id strings
+# windows goodness
 locals {
+  win_builder_instance_type = "t2.large"
+
+  # the one place where win ami key strings (aka "ami keys") are defined
+  win_amis_all_keys = "${split(",","win08,win12,win16,win08pkg,win12pkg,win16pkg")}"
```

Comment:
how about just writing this as a list, rather than interpolating it with the split function?

```
win_amis_all_keys = ["win08","win12","win16","win08pkg","win12pkg","win16pkg"]
```